features/largeVideo/reducer.js

```diff
@@ -9,6 +14,34 @@ const INITIAL_STATE = {
 ReducerRegistry.register('features/largeVideo',
     (state = INITIAL_STATE, action) => {
         switch (action.type) {
+        /**
+         * When conference is joined, we update ID of local participant from
+         * default 'local' to real ID. However in large video we might have
+         * already selected 'local' as participant on stage. So in this case
+         * we must update ID of participant on stage to match ID in
+         * 'participants' state to avoid additional changes in state and
+         * rerenders.
+         */
```

Comment:
I'm afraid I find that design problematic.

A. The whole idea that largeVideo or any feature other than base/participants (i.e. the one which takes care of the change) should be aware of such a thing is a complication & a dependency to me (& hence increases the cost of maintenance, the risk of breakage, etc). If it cannot be avoided, then I'd rather remove details from it i.e. generalize in order to reduce the pieces of the dependency (which I'm going to describe later on in this comment).

B. I find the implementation to know too much & thus increases the dependency problem. It not only knows that there's a change in the id, it knows when it happens (CONFERENCE_JOINED & CONFERENCE_LEFT) & that a specific id is special.

As I said above, I'd rather we reduce the negative effects of the dependency (presuming that the dependency cannot be removed). The fist approach that comes to mind is to:

1. Introduce an action (i.e. event in generic, non-Redux terms) that notifies that a participant's id has changed (& in the well-known format of generic property change events, one may even have the old & new values in the event). That's the very, minimal piece of information that largeVideo cares about. It doesn't care when it happens or whether the involved values are special.

- Then largeVideo (& any other feature for that matter) may subscribe to this 'property change event'  & handle it accordingly without knowing the details.

@tsareg, could I please ask you to fix the problem?